tools/cldr-unittest/src/org/unicode/cldr/unittest/TestLsrvCanonicalizer.java

```diff
@@ -0,0 +1,90 @@
+package org.unicode.cldr.unittest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.unicode.cldr.util.LsrvCanonicalizer;
+import org.unicode.cldr.util.LsrvCanonicalizer.ReplacementRule;
+import org.unicode.cldr.util.LsrvCanonicalizer.TestDataTypes;
+import org.unicode.cldr.util.LsrvCanonicalizer.XLanguageTag;
+import org.unicode.cldr.util.StandardCodes.LstrType;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Objects;
+import com.ibm.icu.dev.test.TestFmwk;
+
+public class TestLsrvCanonicalizer extends TestFmwk {
+
+    static final LsrvCanonicalizer rrs = LsrvCanonicalizer.getInstance();
+    private static final boolean DEBUG = false;
+
+    public static void main(String[] args) {
+        final TestLsrvCanonicalizer testLocaleCanonicalizer = new TestLsrvCanonicalizer();
+        testLocaleCanonicalizer.run(args);
+    }
+
+    public void TestCases() {
+        if (DEBUG) {
+//          System.out.println(Joiner.on('\n').join(rrs.filter(LstrType.variant, null)));
+            System.out.println(Joiner.on('\n').join(rrs.filter(LstrType.language, "no")));
+        }
+
+        List<ReplacementRule> rules = new ArrayList<>();
+        for (Entry<TestDataTypes, Map<String, String>> mainEntry : rrs.getTestData(null).entrySet()) {
+            TestDataTypes type = mainEntry.getKey();
+            for (Entry<String, String> entry : mainEntry.getValue().entrySet()) {
+                String toTest = entry.getKey();
+                String expected = entry.getValue();
+                final XLanguageTag source2 = XLanguageTag.fromTag(LstrType.language, toTest);
+                XLanguageTag newTag = rrs.canonicalizeToX(source2, rules);
+                String actual = newTag.toLocaleString();
+                if (DEBUG && rules.size() > 1) {
+                    System.out.println(
+                        "source: " + toTest
+                        + ", expected: " + expected
+                        + ", actual: " + actual
+                        + ", rules: " + rules
+                        );
+                }
+                if (!Objects.equal(expected, actual)) {
+                    errln("Error: "
+                        + "source: " + toTest
+                        + ", expected: " + expected
+                        + ", actual: " + actual
+                        + ", rules: " + rules
+                        );
+                }
+            }
+        }
+    }
+}
+
+
+//            if (!rule.regular) {
+//                System.out.println(
+//                    rule.lstrType + "\t➕\t" + rule.typeParts + "\t⇒\t"+rule.replacementParts + "\t➕\t" + rule.reason + "\t" + diff(rule.typeParts, rule.replacementParts));
+//            }
+//        }
+//    }
+
+
+
+//    private static String diff(XLanguageTag tagParts, XLanguageTag replacementParts) {
```

Comment:
Adding immediately commented out code without any comments declaring its intent is weird.
I mean yes it's sort of clear it's a debug thing, but there's a DEBUG flag above *and* this is test code anyway, so why not leave it in guarded by the debug flag? At least that way it won't rot over time with successive refactorings of the code around it.

Reply:
It's gone.